micronota/parsers/gff.py

```diff
@@ -0,0 +1,238 @@
+
+Parent
+Indicates the parent of the feature.
+A parent ID can be used to group exons into transcripts, transcripts into genes
+and so forth. A feature may have multiple parents. Parent can *only* be used
+to indicate a partof relationship.
+
+Target
+Indicates the target of a nucleotide-to-nucleotide or
+protein-to-nucleotide alignment. The format of the value is "target_id start
+end [strand]", where strand is optional and may be "+" or "-".
+If the target_id contains spaces, they must be escaped as hex escape %20.
+
+Gap
+The alignment of the feature to the target if the two are not collinear
+(e.g. contain gaps).
+The alignment format is taken from the CIGAR format described in the Exonerate
+documentation.
+
+Derives_from
+Used to disambiguate the relationship between one feature and another when the
+relationship is a temporal one rather than a purely structural "part of" one.
+This is needed for polycistronic genes.
+
+Note
+A free text note.
+
+Dbxref
+A database cross reference. See the GFF3 specification for more information.
+
+Ontology_term
+A cross reference to an ontology term.
+
+Multiple attributes of the same type are indicated by separating
+the values with the comma ","
+
+
+Format Support
+--------------
+**Has Sniffer: Yes**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |No    |:mod:`skbio.metadata.IntervalMetadata` objects                 |
++------+------+---------------------------------------------------------------+
+|Yes   |No    |generator of :mod:`skbio.metadata.IntervalMetadata` objects    |
++------+------+---------------------------------------------------------------+
+
+
+Reference
+---------
+.. [#] http://gmod.org/wiki/GFF3
+'''
+
+from skbio.io import create_format, FileFormatError
+from skbio.metadata import IntervalMetadata, Feature
+from skbio.io.format._base import (
+    _line_generator, _get_nth_sequence, _too_many_blanks)
+
+
+class GFFFormatError(FileFormatError):
+    pass
+
+gff = create_format('gff')
+
+# Annotation headers
+_ANNOTATION_HEADERS = [
+          'SEQID',
+          'SOURCE',
+          'TYPE',
+          'SCORE',
+          'STRAND',
+          'PHASE',
+          'ATTR'
+          ]
+
+# Attribute headers
+_ATTR_HEADERS = [
+        'ID',
+        'Name',
+        'Alias',
+        'Parent',
+        'Target',
+        'Gap',
+        'Derives_from',
+        'Note',
+        'Dbxref',
+        'Ontology_term'
+        ]
+
+
+@gff.sniffer()
+def _gff_sniffer(fh):
+    # check the 1st real line is a valid ID line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        assert line.startswith('##gff-version 3')
```

Comment:
`assert` never raises `GFFFormatError` and `assert` line will be removed if the program runs in optimized mode, so we shouldn't do this... borrowed from SAM parser? I overlooked that when I did review for it.

should do something like this:
```
    if line.startswith('##gff-version 3'):
        return True, {}
    else:
        return False, {}
```